webrender/src/tiling.rs

```diff
@@ -2038,21 +2031,20 @@ impl FrameBuilder {
                         clip_region: &ClipRegion,
                         start_point: LayerPoint,
                         end_point: LayerPoint,
-                        stops: ItemRange) {
-        // Fast paths for axis-aligned gradients:
-        let mut reverse_stops = false;
-        let kind = if start_point.x == end_point.x {
-            GradientType::Vertical
-        } else if start_point.y == end_point.y {
-            GradientType::Horizontal
-        } else {
-            reverse_stops = start_point.x > end_point.x;
-            GradientType::Rotated
-        };
+                        stops: ItemRange,
+                        extend_mode: ExtendMode) {
+        // Fast path for clamped, axis-aligned gradients:
+        let aligned = extend_mode == ExtendMode::Clamp &&
+                      (start_point.x == end_point.x ||
+                       start_point.y == end_point.y);
+        let reverse_stops = !aligned &&
+                            (start_point.x > end_point.x ||
```

Comment:
The important thing here is that if the data is supplied in reverse, that we can produce an equivalent result. What the reference direction is becomes arbitrary, so long as we choose some reference direction. The aligned case didn't have to worry about that because its results rendered actually the same reversed or not, but the new fallback path does need to worry about that in case we have a repeating gradient that would have normally been aligned.